packages/mediapipe-core/lib/src/io/containers.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'package:mediapipe_core/interface.dart';
+import 'package:mediapipe_core/src/io/mediapipe_core.dart';
+import 'third_party/mediapipe/generated/mediapipe_common_bindings.dart'
+    as bindings;
+
+/// {@macro Category}
+// ignore: must_be_immutable
+class Category extends ICategory {
+  /// {@macro Category.fake}
+  Category.fake({
```

Comment:
> This would also enable reusing fakes across the IO and web implementation.

Good point - moving fake constructors into a location that supports this is desirable.

That point aside, I put the constructors here to allow end-user code to more easily test their integration with MediaPipe. My inspiration was the `shared_preferences` library, which [put its testing helpers alongside the main library source](https://github.com/flutter/packages/blob/main/packages/shared_preferences/shared_preferences/lib/shared_preferences.dart#L274). I'm open to other ideas, but am keen to preserve an easy testing story for apps using this library.

Reply:
I think the intention of having publicly available fakes is to allow devs to use them in their own tests, rather then having to write their own fakes.

Rather than including it in the Category class, we usually make a FakeCategory class that extends Category.. would that be better?

Reply:
I spent a while yesterday and this morning exploring different solutions to this problem. I attempted to factor out a `FakeCategory` class, but that ran into challenges as various tiers of the class hierarchy had different ideas about what constructors made sense to declare and, as far as I could tell, _something_ awkward was going to have to leak out _somewhere_.

For now, I've simply dropped the `.fake` name on that constructor and made it the default, unnamed constructor. Separately, in the IO implementation, there is still a named `.native` constructor which accepts a pointer. But, this `.native` named constructor is not visible in normal code completion because it is not declared on the stub file that is exported - only on the native implementation. Later, the web version will add its own special constructor which will also not be visible to surrounding code.

Thus, for a class like `Category`, end users will only see a plain Dart class with a typical, unnamed constructor that accepts raw values. They will be able to construct these freely in their own tests, and in production code will be none the wiser to receiving classes that are mere wrappers around native memory (or, eventually, data that arrived from JS interop).